File: 1290-Convert-Binary-Number-in-a-Linked-List-to-Integer/1290.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        if(head == null) return 0;

        int pow = 0;

        for(ListNode temp = head;
            temp != null;
            temp = temp.next) {
            pow++;
        }

        pow--;

        int retVal = 0;

        while(head != null) {
            retVal += head.val * (1<<pow);
            pow--;
            head=head.next;
        }

        return retVal;

    }
}

File: 1290-Convert-Binary-Number-in-a-Linked-List-to-Integer/1290.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
int getDecimalValue(struct ListNode* head) {
    
    if(head==NULL) return 0;
    if(head->next == NULL) return head->val==1;

    int retVal = 0;
    int pow = 1;

    for(struct ListNode *temp = head;
        temp->next != NULL;
        temp = temp->next)
            pow += 1;

    pow--;
    
    while(head != NULL) {
        retVal += (head->val)*(1<<pow);
        pow--;
        head = head->next;
    }

    return retVal;
}
